File: problems/word-search/boilerplate-full/function.cpp
#include <iostream>
#include <fstream>
#include <vector>
#include <string>
#include <sstream>

##USER_CODE_HERE##

int main() {
  std::ifstream file("/dev/problems/word-search/tests/inputs/##INPUT_FILE_INDEX##.txt");
  std::vector<std::string> lines;
  std::string line;
  while (std::getline(file, line)) lines.push_back(line);

  file.close();
  int outer_size_board;
std::istringstream(lines[0]) >> outer_size_board;
std::vector<std::vector<int>> board(outer_size_board);
for (int i = 0; i < outer_size_board; i++) {
  int inner_size;
  std::istringstream(lines[1 + i]) >> inner_size;
  board[i] = std::vector<int>(inner_size);
  for (int j = 0; j < inner_size; j++) {
    std::istringstream(lines[2 + i]) >> board[i][j];
  }
}
  std::string word;
std::istringstream(lines[1]) >> word;
  bool result = exist(board, word);
  std::cout << result << std::endl;
  return 0;
}

File: problems/word-search/boilerplate-full/function.rs
use std::fs::read_to_string;
use std::io::{self};
use std::str::Lines;

##USER_CODE_HERE##

fn main() -> io::Result<()> {
  let input = read_to_string("/dev/problems/word-search/tests/inputs/##INPUT_FILE_INDEX##.txt")?;
  let mut lines = input.lines();
  let outer_size_board: usize = lines.next().unwrap().parse().unwrap();
let mut board: Vec<Vec<i32>> = Vec::new();
for _ in 0..outer_size_board {
  let inner_size: usize = lines.next().unwrap().parse().unwrap();
  let innerList: Vec<i32> = lines.next().unwrap().split_whitespace().filter_map(|x| x.parse().ok()).collect();
  board.push(innerList);
}
  let word: String = lines.next().unwrap().parse().unwrap();
  let result = exist(board, word);
  println!("{}", result);
  Ok(())
}
